diff --git a/src/main/index.js b/src/main/index.js
index 07ac823a32c6adc2452874d4fe54da0138e58456..8f0c3a801557f441c4dd13574eedfd9e0dcd023c 100644
--- a/src/main/index.js
+++ b/src/main/index.js
@@ -4,7 +4,7 @@ const electronApi = require('./electronApi');
 const { initialize } = require('./initialize');
 const transportConsole = require('./transports/console');
 const transportFile = require('./transports/file');
-const transportRemote = require('./transports/remote');
+// const transportRemote = require('./transports/remote');
 const Logger = require('../core/Logger');
 const ErrorHandler = require('./ErrorHandler');
 
@@ -16,7 +16,7 @@ module.exports = new Logger({
   transportFactories: {
     console: transportConsole,
     file: transportFile,
-    remote: transportRemote,
+    // remote: transportRemote,
   },
   variables: {
     processType: 'main',
