diff --git a/node_modules/@ledgerhq/devices/lib-es/index.js b/node_modules/@ledgerhq/devices/lib-es/index.js
index 3c11191..0146926 100644
--- a/node_modules/@ledgerhq/devices/lib-es/index.js
+++ b/node_modules/@ledgerhq/devices/lib-es/index.js
@@ -11,6 +11,8 @@ var __assign = (this && this.__assign) || function () {
 };
 var _a;
 import semver from "semver";
+import * as _createHIDframing from "./hid-framing";
+export const createHIDframing = _createHIDframing.default
 /**
  * The USB product IDs will be defined as MMII, encoding a model (MM) and an interface bitfield (II)
  *
diff --git a/node_modules/@ledgerhq/devices/lib/index.js b/node_modules/@ledgerhq/devices/lib/index.js
index 74b9d40..7e53a02 100644
--- a/node_modules/@ledgerhq/devices/lib/index.js
+++ b/node_modules/@ledgerhq/devices/lib/index.js
@@ -15,8 +15,10 @@ var __importDefault = (this && this.__importDefault) || function (mod) {
 };
 var _a;
 exports.__esModule = true;
-exports.getInfosForServiceUuid = exports.getBluetoothServiceUuids = exports.identifyProductName = exports.identifyUSBProductId = exports.identifyTargetId = exports.getDeviceModel = exports.ledgerUSBVendorId = exports.DeviceModelId = exports.IIWebUSB = exports.IICCID = exports.IIU2F = exports.IIKeyboardHID = exports.IIGenericHID = void 0;
+exports.getInfosForServiceUuid = exports.getBluetoothServiceUuids = exports.identifyProductName = exports.identifyUSBProductId = exports.identifyTargetId = exports.getDeviceModel = exports.ledgerUSBVendorId = exports.DeviceModelId = exports.IIWebUSB = exports.IICCID = exports.IIU2F = exports.IIKeyboardHID = exports.IIGenericHID = exports.createHIDframing = void 0;
 var semver_1 = __importDefault(require("semver"));
+var hidFraming_1 = __importDefault(require("./hid-framing"));
+exports.createHIDframing = hidFraming_1.default;
 /**
  * The USB product IDs will be defined as MMII, encoding a model (MM) and an interface bitfield (II)
  *
