diff --git a/node_modules/lavamoat/package.json b/node_modules/lavamoat/package.json
index e0993cd..409ba93 100644
--- a/node_modules/lavamoat/package.json
+++ b/node_modules/lavamoat/package.json
@@ -5,7 +5,7 @@
     "lavamoat-run-command": "src/run-command.js"
   },
   "version": "6.4.0",
-  "main": "index.js",
+  "main": "src/index.js",
   "license": "MIT",
   "scripts": {
     "test": "yarn test:prep && yarn test:run",
diff --git a/node_modules/lavamoat/src/index.js b/node_modules/lavamoat/src/index.js
index e304f18..5b762b3 100755
--- a/node_modules/lavamoat/src/index.js
+++ b/node_modules/lavamoat/src/index.js
@@ -34,6 +34,8 @@ async function runLava (options) {
     scuttleGlobalThisExceptions,
     debugMode,
     statsMode,
+    isPackagedApp,
+    includeDevDeps
   } = options
   const shouldParseApplication = writeAutoPolicy || writeAutoPolicyDebug || writeAutoPolicyAndRun
   const shouldRunApplication = (!writeAutoPolicy && !writeAutoPolicyDebug) || writeAutoPolicyAndRun
@@ -43,7 +45,7 @@ async function runLava (options) {
     const includeDebugInfo = Boolean(writeAutoPolicyDebug)
     const policyOverride = await loadPolicy({ debugMode, policyPath: policyOverridePath })
     console.warn(`LavaMoat generating policy from entry "${entryId}"...`)
-    const policy = await parseForPolicy({ projectRoot, entryId, policyOverride, includeDebugInfo })
+    const policy = await parseForPolicy({ projectRoot, entryId, policyOverride, includeDebugInfo, includeDevDeps })
     // write policy debug file
     if (includeDebugInfo) {
       fs.mkdirSync(path.dirname(policyDebugPath), { recursive: true })
@@ -60,7 +62,7 @@ async function runLava (options) {
   if (shouldRunApplication) {
     // execution mode
     const lavamoatPolicy = await loadPolicyAndApplyOverrides({ debugMode, policyPath, policyOverridePath })
-    const canonicalNameMap = await loadCanonicalNameMap({ rootDir: projectRoot, includeDevDeps: true })
+    const canonicalNameMap = await loadCanonicalNameMap({ rootDir: projectRoot, includeDevDeps })
     const kernel = createKernel({
       projectRoot,
       lavamoatPolicy,
@@ -69,6 +71,7 @@ async function runLava (options) {
       scuttleGlobalThisExceptions,
       debugMode,
       statsMode,
+      isPackagedApp
     })

     // run entrypoint
diff --git a/node_modules/lavamoat/src/kernel.js b/node_modules/lavamoat/src/kernel.js
index b65004a..70e8914 100644
--- a/node_modules/lavamoat/src/kernel.js
+++ b/node_modules/lavamoat/src/kernel.js
@@ -14,10 +14,10 @@ const nativeRequire = require

 module.exports = { createKernel }

-function createKernel ({ projectRoot, lavamoatPolicy, canonicalNameMap, debugMode, statsMode, scuttleGlobalThis, scuttleGlobalThisExceptions }) {
+function createKernel ({ projectRoot, lavamoatPolicy, canonicalNameMap, debugMode, statsMode, scuttleGlobalThis, scuttleGlobalThisExceptions, isPackagedApp }) {
   const { resolutions } = lavamoatPolicy
-  const getRelativeModuleId = createModuleResolver({ projectRoot, resolutions, canonicalNameMap })
-  const loadModuleData = createModuleLoader({ canonicalNameMap })
+  const getRelativeModuleId = createModuleResolver({ projectRoot, resolutions, canonicalNameMap, isPackagedApp })
+  const loadModuleData = createModuleLoader({ canonicalNameMap, isPackagedApp })
   const kernelSrc = generateKernel({ debugMode, scuttleGlobalThis, scuttleGlobalThisExceptions })
   const createKernel = evaluateWithSourceUrl('LavaMoat/node/kernel', kernelSrc)
   const reportStatsHook = statsMode ? makeInitStatsHook({ onStatsReady }) : noop
@@ -53,8 +53,16 @@ function prepareModuleInitializerArgs (requireRelativeWithContext, moduleObj, mo
   return [exports, require, module, __filename, __dirname]
 }

-function createModuleResolver ({ projectRoot, resolutions, canonicalNameMap }) {
+function createModuleResolver ({ projectRoot, resolutions, canonicalNameMap, isPackagedApp }) {
   return function getRelativeModuleId (parentAbsolutePath, requestedName) {
+    // allow lavamoat to resolve electron module
+    if (requestedName === 'electron') {
+      return 'electron'
+    }
+
+    if (isPackagedApp && requestedName === 'leveldown') {
+      return 'orbit-db-cache/node_modules/leveldown'
+    }
     // handle resolution overrides
     let parentDir = path.dirname(parentAbsolutePath)
     const parentPackageName = getPackageNameForModulePath(canonicalNameMap, parentDir)
@@ -75,10 +83,10 @@ function createModuleResolver ({ projectRoot, resolutions, canonicalNameMap }) {
   }
 }

-function createModuleLoader ({ canonicalNameMap }) {
+function createModuleLoader ({ canonicalNameMap, isPackagedApp }) {
   return function loadModuleData (absolutePath) {
     // load builtin modules (eg "fs")
-    if (resolve.isCore(absolutePath)) {
+    if (resolve.isCore(absolutePath) || absolutePath === 'electron' || (isPackagedApp && absolutePath === 'orbit-db-cache/node_modules/leveldown')) {
       return {
         type: 'builtin',
         file: absolutePath,
diff --git a/node_modules/lavamoat/src/parseForPolicy.js b/node_modules/lavamoat/src/parseForPolicy.js
index 286e633..cfd1509 100644
--- a/node_modules/lavamoat/src/parseForPolicy.js
+++ b/node_modules/lavamoat/src/parseForPolicy.js
@@ -55,10 +55,10 @@ module.exports = {
   resolutionOmittedExtensions
 }

-async function parseForPolicy ({ projectRoot, entryId, policyOverride = {}, rootPackageName, shouldResolve, includeDebugInfo, ...args }) {
+async function parseForPolicy ({ projectRoot, entryId, policyOverride = {}, rootPackageName, shouldResolve, includeDebugInfo, includeDevDeps, ...args }) {
   const isBuiltin = (id) => builtinPackages.includes(id)
   const { resolutions } = policyOverride
-  const canonicalNameMap = await loadCanonicalNameMap({ rootDir: projectRoot, includeDevDeps: true })
+  const canonicalNameMap = await loadCanonicalNameMap({ rootDir: projectRoot, includeDevDeps })
   const resolveHook = makeResolveHook({ projectRoot, resolutions, rootPackageName, canonicalNameMap })
   const importHook = makeImportHook({ rootPackageName, shouldResolve, isBuiltin, resolveHook, canonicalNameMap })
   const moduleSpecifier = resolveHook(entryId, path.join(projectRoot, `package.json`))
@@ -107,7 +107,7 @@ function makeImportHook ({
 }) {
   return async (specifier) => {
     // see if its a builtin
-    if (isBuiltin(specifier)) {
+    if (isBuiltin(specifier) || specifier.includes('node_modules/electron/index.js')) {
       return makeBuiltinModuleRecord(specifier)
     }

diff --git a/node_modules/lavamoat/src/yargsFlags.js b/node_modules/lavamoat/src/yargsFlags.js
index 9617ac8..138e467 100644
--- a/node_modules/lavamoat/src/yargsFlags.js
+++ b/node_modules/lavamoat/src/yargsFlags.js
@@ -60,5 +60,12 @@ module.exports = (yargs, defaults) => {
     describe: 'enable writing and logging of stats',
     type: 'boolean',
     default: defaults.statsMode
+  }),
+  // log initialization stats
+  yargs.option('includeDevDeps', {
+    alias: ['includeDevDeps'],
+    describe: 'include dev deps when getting deps canonical name map',
+    type: 'boolean',
+    default: true
   })
 }
\ No newline at end of file
