diff --git a/node_modules/lavamoat/node_modules/lavamoat-core/src/loadPolicy.js b/node_modules/lavamoat/node_modules/lavamoat-core/src/loadPolicy.js
index 05958c9..4be44b2 100644
--- a/node_modules/lavamoat/node_modules/lavamoat-core/src/loadPolicy.js
+++ b/node_modules/lavamoat/node_modules/lavamoat-core/src/loadPolicy.js
@@ -27,10 +27,13 @@ async function loadPolicyAndApplyOverrides({ debugMode, policyPath, policyOverri
     if (debugMode) console.warn(`Merging policy-override.json into policy.json`)
     const policyOverride = await readPolicyFile({ debugMode, policyPath: policyOverridePath })
     lavamoatPolicy = mergePolicy(policy, policyOverride)
-    // TODO: Only write if merge results in changes.
-    // Would have to make a deep equal check on whole policy, which is a waste of time.
-    // mergePolicy() should be able to do it in one pass.
-    fs.writeFileSync(policyPath, jsonStringify(lavamoatPolicy, { space: 2 }))
+
+    if (JSON.stringify(policy) !== JSON.stringify(lavamoatPolicy)) {
+      // TODO: Only write if merge results in changes.
+      // Would have to make a deep equal check on whole policy, which is a waste of time.
+      // mergePolicy() should be able to do it in one pass.
+      fs.writeFileSync(policyPath, jsonStringify(lavamoatPolicy, { space: 2 }))
+    }
   }
   return lavamoatPolicy;
 }
