diff --git a/node_modules/lavamoat/node_modules/lavamoat-core/src/loadPolicy.js b/node_modules/lavamoat/node_modules/lavamoat-core/src/loadPolicy.js
index ef71923..34e901e 100644
--- a/node_modules/lavamoat/node_modules/lavamoat-core/src/loadPolicy.js
+++ b/node_modules/lavamoat/node_modules/lavamoat-core/src/loadPolicy.js
@@ -36,7 +36,7 @@ async function loadPolicyAndApplyOverrides({ debugMode, policyPath, policyOverri
     // TODO: Only write if merge results in changes.
     // Would have to make a deep equal check on whole policy, which is a waste of time.
     // mergePolicy() should be able to do it in one pass.
-    fs.writeFileSync(policyPath, jsonStringify(lavamoatPolicy, { space: 2 }))
+    //fs.writeFileSync(policyPath, jsonStringify(lavamoatPolicy, { space: 2 }))
   }
   return lavamoatPolicy
 }
